cmake/install_configs.cmake

```diff
@@ -0,0 +1,33 @@
+#######################
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+#  agreements.  See the NOTICE file distributed with this work for additional information regarding
+#  copyright ownership.  The ASF licenses this file to you under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with the License.  You may obtain
+#  a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing permissions and limitations under
+#  the License.
+#
+#######################
+
+# This is a script intended to be passed to install(SCRIPT ...)
+# This copies default config files to the destination  without overwriting existing files.
+# If the config file already exists, the source config is copied with the '.default' extension
+
+file(GLOB CONFIG_FILES ${CONFIG_SOURCE_GLOBS})
+file(MAKE_DIRECTORY ${CONFIG_DEST_PATH})
+foreach(CONFIG_FILE ${CONFIG_FILES})
+  cmake_path(GET CONFIG_FILE STEM LAST_ONLY CONFIG_FILE_NAME)
```

Comment:
Should we do a sanity check to make sure `CONFIG_FILE_NAME` isn't an empty string?

> If a requested component is not present in the path, an empty string will be stored in <out-var>.

I'm pretty sure the current implementation will error out later if this happens, but nobody wants the config directory to be accidentally wiped in an unexpected edge case.

Reply:
I'm not sure they can be empty based on the glob of "*.default"

Reply:
Probably safe then. And I did think through it more carefully and I am very confident that even if it is empty it will not do anything crazy.